Update component “dropdown-multi-selects” (#153)
Zywave/booster: src/_areas/design-system/components/dropdown-multi-selects.md

* A Dropdown multi-select shouldn't default a selected option. Defaulting a selection only makes sense after a user's selection has been saved and they revisit to update their choice.
* Sometimes choosing "None" as an option is necessary if no options are applicable to the user when a Dropdown multi-select is required. "None" should appear at the top of the list.

#### Selection wrapping and truncation

> * When options are selected from the Dropdown multi-select option list, selections will populate inside the select box. Multiple selections should wrap within the select box to fit all.  

Reviewer: ```suggestion
* When options are selected from the Dropdown multi-select option list, they will populate inside the select box as selection tags. Multiple selection tags should wrap within the select box to fit all.  
```

![Tags selection wrapping](/images/tag_specs.svg)

> * In extreme circumstances, truncation can be used to prevent the selection container from continually expanding. 

Reviewer: ```suggestion
* In extreme circumstances, truncation can be used to prevent the select box from continually expanding. 
```

>   * We recommend displaying a maximum of 5 selection tags, and a sixth that summarizes how many other options are selected (e.g. "and 10 more").

Reviewer: ```suggestion
  * We recommend displaying a maximum of 5 selection tags, and a sixth that summarizes how many other options are selected (e.g., "and 10 more").
```

* A Dropdown multi-select shouldn't default a selected option. Defaulting a selection only makes sense after a user's selection has been saved and they revisit to update their choice.
* Sometimes choosing "None" as an option is necessary if no options are applicable to the user when a Dropdown multi-select is required. "None" should appear at the top of the list.

#### Selection wrapping and truncation

> * When options are selected from the option list, they will populate inside the select form field as select tags. Multiple select tags should wrap within the select form field to fit all.

Author: In the "Anatomy" section, we call this part of the select dropdown "selection tags". We should align these terms, I think. But I'm not invested in either choice.

  * Backspacing over a selection in the select box removes it.
  * Deselecting the checkbox within the option list will also remove the selection.

#### Select all

> In some situations, a "Select all" option is desirable (e.g. "Select all accounts" when sending an email or "Select all states" when managing prefrences). There are some alternate behaviors to consider; choose what is most appropriate for your use case:

Reviewer: ```suggestion
In some situations, a "Select all" option is desirable (e.g., "Select all accounts" when sending an email or "Select all states" when managing preferences). There are some alternate behaviors to consider; choose what is most appropriate for your use case:
```

> 1. Select all to select all options present in the UI:

Reviewer: ```suggestion
1. Select all to choose every option present in the UI:
```

>     * If the Multi-select dropdown has a finite list of options (e.g. 50 states), the "Select all" option can simply select all of these known options, so that the user sees an individual selection tag for each option

Reviewer: ```suggestion
    * If the Dropdown multi-select has a finite list (e.g., 50 states), the "Select all" option can simply select everything, so that the user sees an individual selection tag for each in the select box.
```
Trying to limit the amount of times "option" was used, lol.

    * A user can deselect options as normal. The "Select all" option is checked in the "Option list" as long as all other options are selected.

>     * For example: a user is presented with a Multi-select dropdown for the 50 states. They choose to "Select all". 50 selection tags are then displayed in the selection container.

Reviewer: ```suggestion
    * For example: a user is presented with a Dropdown multi-select for 50 states. They choose to "Select all" and 50 selection tags are then displayed in the select box.
```

> 1. Select all to select all options present in a data store (e.g. many accounts):

Reviewer: ```suggestion
1. Select all to choose every option present in a data store (e.g., many accounts):
```

>     * If the Multi-select dropdown is backed by an API to yield an unknown and potentially large quantity of options, the "Select all" option may need to be optimized to handle this load. In this situation, "Select all" may need to be preserved as a magic value, to be interpreted by the application in more robust ways.

Reviewer: ```suggestion
    * If the Dropdown multi-select is backed by an API to yield an unknown and potentially large quantity of options, the "Select all" option may need to be optimized to handle this load. In this situation, "Select all" may need to be preserved as a magic value, to be interpreted by the application in more robust ways.
```

>     * A user cannot deselect individual selections after they have indicated "Select all". They must explicitly deselect "Select all", which will require them to select the the individual options they want.

Reviewer: ```suggestion
    * A user cannot deselect individual selections after they have chosen the "Select all" option. They must explicitly deselect "Select all", which will require them to select the individual options they want.
```

>     * For example: a user is presented with a UI to send an email to their clients. They choose "Select all" as recipients. An "All selected" selection tag is displayed in the selection container.

Reviewer: ```suggestion
    * For example: a user is presented with a UI to send an email to their clients. They choose "Select all" as recipients. An "All selected" selection tag is displayed in the select box.
```
